math/binomialcoefficient_test.go

```diff
@@ -0,0 +1,51 @@
+// binomialcoefficient_test.go
+// description: Returns C(n, k)
+// author: Akshay Dubey (https://github.com/itsAkshayDubey)
+// see binomialcoefficient.go
+
+package math_test
+
+import (
+	"errors"
+	"strings"
+	"testing"
+
+	"github.com/TheAlgorithms/Go/math"
+)
+
+func TestC(t *testing.T) {
+	var tests = []struct {
+		n        int
+		k        int
+		expected int
+	}{
+		{5, 2, 10},
+		{7, 4, 35},
+		{0, 0, 1},
+		{-1, 1, -1},
+		{1, -1, -1},
+		{-1, -1, -1},
+	}
+	for _, test := range tests {
+		var expectedError error = errors.New("arguments must be positive")
+		result, error := math.C(test.n, test.k)
+		t.Log(test.n, " ", result, " ", error)
+		if result != test.expected {
+			t.Errorf("Wrong result! Expected:%v, Returned:%v", test.expected, result)
+		}
+		if result == -1 && error == nil {
+			t.Errorf("Wrong result! Expected:%v, Returned:%v", expectedError, error.Error())
+		}
+		if result != -1 && error != nil {
+			t.Errorf("Wrong result! Expected:%v, Returned:%v", nil, error.Error())
+		}
+		if result == -1 && !strings.EqualFold(expectedError.Error(), error.Error()) {
+			t.Errorf("Wrong result! Expected:%v, Returned:%v", expectedError.Error(), error.Error())
+		}
+	}
```

Comment:
Put this under a `t.Run` block.


For example
```Go
t.Run(name, func(t *testing.T) {
            got := Split(tc.input, tc.sep)
            if !reflect.DeepEqual(tc.want, got) {
                t.Fatalf("expected: %v, got: %v", tc.want, got)
            }
        })
```

In this case,

```suggestion
		t.Run(name, func(t *testing.T) {
			result, err := math.C(test.n, test.k)
			if result != test.expected || test.expectedErr != err {
				t.Errorf("expected error: %s, got: %s; expected value: %v, got: %v", test.expectedErr, err, test.expectedErr)
			}	
        	})
	}
```

The above is more than enough - no need for `t.Log` and 4 different cases when one suffices.

This also means that you would have to update your test case struct to include `expectedErr` field and use the `math.ErrPosArgsOnly` (after you have taken it out of your function and defined it inside the package instead) on the ones you expect to fail rather than creating a new `error` and checking `string` values.